resources/replace_or_add.rb

```diff
@@ -1,12 +1,12 @@
 property :path, String
-property :pattern, String
+property :pattern, [String, Regexp]
 property :line, String
 property :replace_only, [true, false]
 
 resource_name :replace_or_add
 
 action :edit do
-  regex = /#{new_resource.pattern}/
+  regex = new_resource.pattern.is_a?(String) ? /#{new_resource.pattern}/ : new_resource.pattern
```

Comment:
nice

Reply:
I'm just standardising that pattern where I can. If/when we remove it, it'll be the same change rather than guessing who, what, why.